src/mlpack/methods/ann/image_functions/bilinear_function.hpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file bilinear_function.hpp
+ * @author Kris Singh
+ *
+ * Definition and implementation of the bilinear interpolation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_BILINEAR_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_IMAGE_FUNCTIONS_BILINEAR_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <assert.h>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The bilinear interpolation function
+ *
+ */
+class BiLinearFunction
+{
+ /*
+  * The constructor for the bilinear interpolation.
+  * 
+  * @param inRowSize: Number of input rows
+  * @param inColSize: Number of input columns.
+  * @param outRowSize: Number of output rows.
+  * @param outColSie: Number of output columns.
+  */
+ public:
+  BiLinearFunction(
+      const size_t inRowSize,
+      const size_t inColSize,
+      const size_t outRowSize,
+      const size_t outColSize,
+      const size_t depth = 1):
+      inRowSize(inRowSize),
+      inColSize(inColSize),
+      outRowSize(outRowSize),
+      outColSize(outColSize),
+      depth(depth)
+  {};
+  /**
+   * UpSample the given input.
+   *
+   * @param input The input matrix
+   * @param output The resulting interpolated output image.
+   */
+  template<typename eT>
+  void UpSample(const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  {
+    if (output.is_empty())
+      output.set_size(outRowSize * outColSize, 1);
```

Comment:
That's incorrect. You should write
`output.set_size(outRowSize * outColSize * depth, 1);`